menas/src/test/scala/za/co/absa/enceladus/menas/integration/repositories/StatisticsIntegrationSuite.scala

```diff
@@ -0,0 +1,116 @@
+      Map("mandatoryString1"->"","mandatoryString2"->"3"))),
+    DatasetFactory.getDummyDataset(name = "dataset4", version = 1, properties = Some(
+      Map("mandatoryString1"->"", "mandatoryString2"->"3", "recommendedString1" -> "", "optionalString1"->"",
+        "mandatoryDisabledString1" -> "", "optionalEnumAb" -> "optionA"))),
+    DatasetFactory.getDummyDataset(name = "dataset5", version = 1, properties = Some(Map())),
+    DatasetFactory.getDummyDataset(name = "dataset6", version = 1, properties = Some(Map(
+      "mandatoryString1"->"", "mandatoryString2"->"3", "recommendedString1" -> "")))
+  )
+
+  val mockedPropertyDefinitions = Seq(
+    PropertyDefinition(name = "mandatoryString1", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
+      userCreated = "", userUpdated = ""),
+    PropertyDefinition(name = "mandatoryString2", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
+      userCreated = "", userUpdated = ""),
+    PropertyDefinition(name = "recommendedString1", propertyType = StringPropertyType(), essentiality = Recommended,
+      userCreated = "", userUpdated = ""),
+    PropertyDefinition(name = "optionalString1", propertyType = StringPropertyType(), essentiality = Optional,
+      userCreated = "", userUpdated = ""),
+    PropertyDefinition(name = "mandatoryDisabledString1", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
+      disabled = true, userCreated = "", userUpdated = ""),
+    PropertyDefinition(name = "optionalEnumAb", propertyType = EnumPropertyType("optionA", "optionB"), essentiality = Optional,
+      userCreated = "", userUpdated = "")
+  )
+
+  "StatisticsService" should {
+
+    "return the properties with missing counts" when {
+      "the specified datasets and propertie" in {
+        datasetFixture.add(mockedDatasets: _*)
+        propertyDefService.add(mockedPropertyDefinitions: _*)
+
+        val actualStatistics = await(statisticsService.getPropertiesWithMissingCount())
+
+        val expectedStatistics = Seq(
+          PropertyDefinitionDto(name = "mandatoryString1", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
+            datasetNumberMissing = 1),
+          PropertyDefinitionDto(name = "mandatoryString2", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
+            datasetNumberMissing = 2),
+          PropertyDefinitionDto(name = "optionalEnumAb", propertyType = EnumPropertyType("optionA", "optionB"), essentiality = Optional,
+            datasetNumberMissing = 4),
+          PropertyDefinitionDto(name = "optionalString1", propertyType = StringPropertyType(), essentiality = Optional,
+            datasetNumberMissing = 4),
+          PropertyDefinitionDto(name = "recommendedString1", propertyType = StringPropertyType(), essentiality = Recommended,
+            datasetNumberMissing = 3)
+        )
```

Comment:
```suggestion
        val expectedStatistics = Seq(
          PropertyDefinitionDto(name = "mandatoryString1", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false),
            datasetNumberMissing = 1), // missing in dataset5
          PropertyDefinitionDto(name = "mandatoryString2", propertyType = StringPropertyType(), essentiality = Mandatory(allowRun = false), 
            datasetNumberMissing = 2),  // missing in dataset5
          PropertyDefinitionDto(name = "optionalEnumAb", propertyType = EnumPropertyType("optionA", "optionB"), essentiality = Optional,
            datasetNumberMissing = 4),  // missing in dataset2,3,5,6
          PropertyDefinitionDto(name = "optionalString1", propertyType = StringPropertyType(), essentiality = Optional,
            datasetNumberMissing = 4), // dataset1,3,5,6
          PropertyDefinitionDto(name = "recommendedString1", propertyType = StringPropertyType(), essentiality = Recommended,
            datasetNumberMissing = 3) // dataset1,3,5
        )
```

Again, being easier on the reader to reason about.